plaso/parsers/vsftpd.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+"""Parser for vsftpd Logs."""
+
+from __future__ import unicode_literals
+
+import calendar
+
+import pyparsing
+from dfdatetime import time_elements as dfdatetime_time_elements
+
+from plaso.containers import events
+from plaso.containers import time_events
+from plaso.lib import errors
+from plaso.lib import definitions
+from plaso.parsers import logger
+from plaso.parsers import text_parser
+from plaso.parsers import manager
+
+
+class VsftpdEventData(events.EventData):
+  """vsftpd Log event data.
+
+  Attributes:
+    text (str): vsftpd log message.
+  """
+
+  DATA_TYPE = 'vsftpd:log'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(VsftpdEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.text = None
+
+
+class VsftpdLogParser(text_parser.PyparsingSingleLineTextParser):
+  """Parses the vsftpd log."""
+
+  NAME = 'vsftpd'
+  DESCRIPTION = 'Parser for vsftpd log files.'
+
+  _DATETIME_ELEMENTS = (
+      text_parser.PyparsingConstants.THREE_LETTERS.setResultsName('dow') +
+      text_parser.PyparsingConstants.THREE_LETTERS.setResultsName('month') +
+      text_parser.PyparsingConstants.ONE_OR_TWO_DIGITS.setResultsName(
+          'day_of_month') +
+      text_parser.PyparsingConstants.TWO_DIGITS.setResultsName('hours') +
+      pyparsing.Suppress(':') +
+      text_parser.PyparsingConstants.TWO_DIGITS.setResultsName('minutes') +
+      pyparsing.Suppress(':') +
+      text_parser.PyparsingConstants.TWO_DIGITS.setResultsName('seconds') +
+      text_parser.PyparsingConstants.FOUR_DIGITS.setResultsName('year'))
+
+  # Note that the whitespace is suppressed by pyparsing.
+  _DATE_TIME = pyparsing.Group(_DATETIME_ELEMENTS)
+
+  _LOG_LINE = (
+      _DATE_TIME.setResultsName('date_time') +
+      pyparsing.SkipTo(pyparsing.lineEnd).setResultsName('text'))
+
+  LINE_STRUCTURES = [
+      ('logline', _LOG_LINE),
+  ]
+
+  def _get_time_elements_tuple(self, structure):
```

Comment:
This needs to be _GetTimeElementsTuple, per the style guide. Please also add Args and returns sections to the docstring

Reply:
I've made changes as requested and tuned code to be consistent with other modules as well (pass full structure to the func, use timelib instead of calendar)